plumbing/util.go

```diff
@@ -179,36 +177,27 @@ func CloneRepository(remote GitRemote, inMem bool, basePath string) (rep *git.Re
 	return
 }
 
-// IsBinary detects if data is a binary value based on:
-// http://git.kernel.org/cgit/git/git.git/tree/xdiff-interface.c?id=HEAD#n198
+// IsBinary returns whether a file contains a NUL byte near the beginning of the file.
 func IsBinary(filePath string) (bool, error) {
 	r, err := os.Open(filePath)
 	if err != nil {
 		return false, err
 	}
 	defer r.Close()
 
-	reader := bufio.NewReader(r)
-	c := 0
-	for {
-		if c == SNIFFLEN {
-			break
-		}
-
-		b, err := reader.ReadByte()
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			return false, err
+	buf := make([]byte, 8000)
```

Comment:
Why do you tend to replace constants with literals?

Reply:
I try to scope constants down to as close to where they are needed as possible. Because this is used once in a single function and the number is easy to understand in context, I inlined. If the number had been needed multiple times inside the same function, I would have created a named `const` inside the function scope.